integration-tests/src/lib.rs

```diff
@@ -19,96 +19,95 @@ pub struct TestContext {
+            Err(e) => {
+                panic!("Error running vexination indexer: {:?}", e);
+            }
+            Ok(code) => {
+                println!("Vexination indexer exited with code {:?}", code);
+            }
+        },
 
-            bapi = bombastic_api(&api.storage_endpoint).run() => match bapi {
-                Err(e) => {
-                    panic!("Error running bombastic API: {:?}", e);
-                }
-                Ok(code) => {
-                    println!("Bombastic API exited with code {:?}", code);
-                }
-            },
 
-            vapi = vexination_api(&api.storage_endpoint).run() => match vapi {
-                Err(e) => {
-                    panic!("Error running vexination API: {:?}", e);
-                }
-                Ok(code) => {
-                    println!("Vexination API exited with code {:?}", code);
-                }
-            },
-
-            _ = async move {
-                let client = reqwest::Client::new();
-                // Probe bombastic API
-                loop {
-                    let response = client
-                        .get("http://localhost:8082/api/v1/sbom?id=none")
-                        .send()
-                        .await
-                        .unwrap();
-                    if response.status() == StatusCode::NOT_FOUND {
-                        break;
-                    }
-                    tokio::time::sleep(Duration::from_secs(1)).await;
-                }
+        bapi = server => match bapi {
+            Err(e) => {
+                panic!("Error running bombastic API: {:?}", e);
+            }
+            Ok(code) => {
+                println!("Bombastic API exited with code {:?}", code);
+            }
+        },
 
-                // Probe vexination API
-                loop {
-                    let response = client
-                        .get("http://localhost:8081/api/v1/vex?advisory=none")
-                        .send()
-                        .await
-                        .unwrap();
-                    if response.status() == StatusCode::NOT_FOUND {
-                        break;
-                    }
-                    tokio::time::sleep(Duration::from_secs(1)).await;
-                }
+        vapi = vexination_api(&api.storage_endpoint).run() => match vapi {
+            Err(e) => {
+                panic!("Error running vexination API: {:?}", e);
+            }
+            Ok(code) => {
+                println!("Vexination API exited with code {:?}", code);
+            }
+        },
 
-                // Run test
-                test.await
-            } => {
-                println!("Test completed");
+        _ = async move {
+            let client = reqwest::Client::new();
+            // Probe bombastic API
+            loop {
+                let response = client
+                    .get(format!("http://localhost:{}/api/v1/sbom?id=none", bapi.port))
```

Comment:
@lulf I figured out how to make the OS-assigned port available to this section, but I've yet to figure out how to pass it to the test itself. I feel like I'm missing something simple. :shrug:

Reply:
Nevermind -- I figured it out.